website/src/content/docs/docs/libraries/http/operations.md

```diff
@@ -103,6 +105,165 @@ namespace Pets {
 }
 ```
 
+### `@body` vs `@bodyRoot`
+
+The `@body` decorator defines the property which type will be exactly the body. Annotating a property inside a `@body` with applicable metadata (`@header`, `@path`, `@query` for request and `@header`, `@statusCode` for response) will be ignored and log a warning.
+The `@bodyRoot` decorator defines the property where the implicit body resolution will be applied(By default the root of the operation).
+
+Nesting `@body` and `@bodyRoot` while mostly pointless can happen when using templates to build operations. A warning will be emitted if nesting happens inline.
+The meaning when nesting them is as follow:
+
+- As soon as `@body` is reached the content is exactly the body which means any nested `@bodyRoot` or `@body` will be ignored.
+- When reaching a `@bodyRoot` it will keep looking for `@body` and `@bodyRoot` nested and if found the deepest one will be used as the body.
```

Comment:
```suggestion
- if `@bodyRoot` is reached before any occurrence of `@body`, it will keep looking for nested properties decorated with  `@body` or `@bodyRoot` and if found the deepest one will be used to determine the body.
```